### PR TITLE
Feature/Add External Initiators feature flag

### PR DESCRIPTION
**Pull request by felder-cl**
Introduce the configuration flag 'FeatureExternalInitiators' (FEATURE_EXTERNAL_INITIATORS) to enable using external initiators.

https://www.pivotaltracker.com/n/projects/2129823/stories/168647972

### CODE REVIEW

File: core/web/external_initiators_controller.go [hunk 1]
```diff
@@ -20,8 +20,11 @@ type ExternalInitiatorsController struct {
 // Create builds and saves a new service agreement record.
 func (eic *ExternalInitiatorsController) Create(c *gin.Context) {
 	eir := &models.ExternalInitiatorRequest{}
-	if !eic.App.GetStore().Config.Dev() {
-		jsonAPIError(c, http.StatusMethodNotAllowed, errors.New("External Initiators are currently under development and not yet usable outside of development mode"))
+	if !(eic.App.GetStore().Config.Dev() || eic.App.GetStore().Config.FeatureExternalInitiators()) {
```
Comment (j16r) [inline core/web/external_initiators_controller.go]: I feel like the contrapositive would be easier to read here.

Comment (felder-cl) [reply core/web/external_initiators_controller.go]: If you think of it as an 'unless' statement its easier.  The contrapositive would use a series of not's and 'ands' and its more to process in my opinion.  Anyone else?

Comment (dimroc) [reply core/web/external_initiators_controller.go]: I do prefer `if` statements in the positive and am not a fan of `unless` in ruby. This line is taking me a while to grok. We want both to be false, correct? That does seem easier to understand:

```
if !(eic.App.GetStore().Config.Dev() || eic.App.GetStore().Config.FeatureExternalInitiators())

if eic.App.GetStore().Config.Dev() == False && eic.App.GetStore().Config.FeatureExternalInitiators() == False

if !eic.App.GetStore().Config.Dev() && !eic.App.GetStore().Config.FeatureExternalInitiators()
```

**Comment by felder-cl** [comment]
:+1:

**Pull request merged**